test/Microsoft.Azure.Devices.Edge.Test/ManifestSigning.cs

```diff
@@ -0,0 +1,222 @@
+                                .WithEnvironment(new[] { ("MessageCount", "-1") });
+                        },
+                        manifestSigningCts.Token,
+                        Context.Current.NestedEdge,
+                        inputManifestSettings);
+                    Console.WriteLine("Deployment done ");
+                }
+                catch (TaskCanceledException)
+                {
+                    Console.WriteLine("The task got cancelled - hello");
+                }
+
+                CancellationTokenSource getTwinTimer = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+                Twin twin = await this.IotHub.GetTwinAsync(this.runtime.DeviceId, Option.Some("$edgeAgent"), getTwinTimer.Token);
+                Assert.AreNotEqual(twin.Properties.Desired.Version, twin.Properties.Reported.GetLastUpdatedVersion());
+            }
+        }
+
+        /*[Category("ManifestSigning")]
+        [Test]
+        public async Task TestIfSignedDeploymentIsConfiguredWithNoRootCa()
+        {
+            // Edge Daemon is not configured but manifest is signed.
+            this.SetLaunchSettingsWithRootCa(Context.Current.ManifestSigningDefaultLaunchSettings, Context.Current.ManifestSigningGoodRootCaPath);
+            ManifestSettings inputManifestSettings = new ManifestSettings(Context.Current.ManifestSigningDeploymentPath, Context.Current.ManifestSigningSignedDeploymentPath, Context.Current.ManifestSigningGoodRootCaPath, Context.Current.ManifestSignerClientDirectory, Context.Current.ManifestSignerClientProjectPath);
+
+            await this.SetConfigToEdgeDaemon(Option.None<string>(), this.TestToken);
+
+            // Set a faster time out as its expected to fail.
+            CancellationTokenSource manifestSigningCts = new CancellationTokenSource(TimeSpan.FromMinutes(2));
+
+            // This is a temporary solution see ticket: 9288683
+            if (!Context.Current.ISA95Tag)
+            {
+                try
+                {
+                    EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(
+                        builder =>
+                        {
+                            builder.AddModule(SensorName, this.sensorImage)
+                                .WithEnvironment(new[] { ("MessageCount", "-1") });
+                        },
+                        manifestSigningCts.Token,
+                        Context.Current.NestedEdge,
+                        inputManifestSettings);
+                }
+                catch (TaskCanceledException)
+                {
+                    CancellationTokenSource getTwinTimer = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+                    Twin twin = await this.IotHub.GetTwinAsync(this.runtime.DeviceId, Option.Some("$edgeAgent"), getTwinTimer.Token);
+                    Assert.AreNotEqual(twin.Properties.Desired.Version, twin.Properties.Reported.GetLastUpdatedVersion());
```

Comment:
Like in the test above, we should move this out of the test block.

Reply:
Also, I wonder if we should put a comment here:
// EdgeAgent will reject the new deployment, so the reported and desired versions will not match,